test/Test/Indigo/Contracts/HomebaseLite/Vote/Entrypoints.hs

```diff
@@ -0,0 +1,135 @@
+-- SPDX-FileCopyrightText: 2022 Tezos Commons
+-- SPDX-License-Identifier: LicenseRef-MIT-TC
+
+{-# LANGUAGE OverloadedLists, PartialTypeSignatures #-}
+{-# OPTIONS_GHC -Wno-partial-type-signatures #-}
+
+module Test.Indigo.Contracts.HomebaseLite.Vote.Entrypoints
+  ( test_propose
+  , test_vote
+  ) where
+
+import Lorentz (CustomError, IsError, MText, MustHaveErrorArg)
+
+import Data.Default
+import Test.Tasty (TestTree, testGroup)
+
+import qualified Indigo.Contracts.FA2Sample as FA2
+import qualified Lorentz.Contracts.Spec.FA2Interface as FA2
+import Morley.Tezos.Core
+import Morley.Util.Label
+import Morley.Util.Named
+import Test.Cleveland
+import Test.Cleveland.Lorentz.Types (toAddress)
+
+import Indigo.Contracts.HomebaseLite
+import Indigo.Contracts.HomebaseLite.Types
+import Test.Indigo.Contracts.HomebaseLite.AsRPC
+import Test.Indigo.Contracts.HomebaseLite.Utils
+
+test_propose :: TestTree
+test_propose = testGroup "propose entrypoint"
+  [ testScenario "works" $ scenario do
+      (holder, _, contract) <- prepare
+      lastLevel <- getLevel
+      timestamp <- getNow
+      withSender holder do
+        call contract (Call @"Propose") (uri, #choices :! choices)
+      storage <- getStorage contract
+      let config = sConfigurationRPC storage
+          proposals = sProposalsRPC storage
+      proposal <- getBigMapValue proposals uri
+      piChoices proposal @== choices
+      piQuorumThreshold proposal @== cQuorumThresholdRPC config
+      -- this is not a very precise test, but can't really do much better
```

Comment:
:thinking: indeed, TBH I wouldn't expect this to succeed consistently on the testnet, due to the variation in the actual time that it takes to submit the operation.

Perhaps we could change these in checks that are not as strict, or use some other woraround

Reply:
Uh. The checks are currently lax as all hell, those basically only test that both start and expiration time are not in the past relative to the origination time, same for the level. Exactly because the more strict tests would randomly fail on the testnet. FWIW, on the emulator those could be much more precise. Once morley!1044 is merged (hopefully eventually) we'll be able to define different behaviour for net/emulator test inside the same scenario. Until then, eh, perhaps either leave those as-is or remove altogether. I'll add a todo to that effect. That said, property tests in #19 also test for basically the same thing.

Reply:
:facepalm: somehow I though the test were checking for strict equality, my bad.

Yes, that's about as good as we can get right now, disregard my previous comment